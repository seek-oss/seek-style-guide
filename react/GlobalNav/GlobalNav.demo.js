import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import GlobalNav from './GlobalNav';

export const makeDummyLinkRendererForPath = path => {
  const DummyLinkRenderer = ({ href, ...props }) => (
    <Link to={`${path}#actualHref=${href}`} {...props} />
  );

  DummyLinkRenderer.propTypes = {
    href: PropTypes.string
  };

  return DummyLinkRenderer;
};

const ROUTE = '/global-nav';

export default {
  route: ROUTE,
  title: 'Global Nav',
  category: 'Layout',
  component: GlobalNav,
  initialProps: {
    locale: 'AU',
    linkRenderer: makeDummyLinkRendererForPath(ROUTE),
    hideLocales: false
  },
  options: [
    {
      label: 'Locale States',
      type: 'checklist',
      states: [
        {
          label: 'Hide locales',
          transformProps: props => ({
            ...props,
            hideLocales: true
          })
        },
        {
          label: 'Switch to NZ site',
          transformProps: props => ({
            ...props,
            locale: 'NZ'
          })
        }
      ]
    },
    {
      label: 'Active Partner Site',
      type: 'radio',
      states: [
        null,
        'jobs',
        'courses',
        'businesses',
        'volunteering'
      ].map(activePartnerSite => ({
        label: activePartnerSite || 'No active partner',
        transformProps: props => ({
          ...props,
          activePartnerSite
        })
      }))
    }
  ]
};
