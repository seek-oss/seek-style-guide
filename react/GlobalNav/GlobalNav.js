import styles from './GlobalNav.less';
import React from 'react';
import PropTypes from 'prop-types';

import Hidden from '../Hidden/Hidden';
import Locales from './Locales/Locales';
import PartnerSites from './PartnerSites/PartnerSites';

export default function GlobalNav({ locale, linkRenderer, hideLocales, activePartnerSite }) {
  return (<Hidden print mobile className={styles.root}>
    <div className={styles.content}>
      <PartnerSites locale={locale} linkRenderer={linkRenderer} activePartnerSite={activePartnerSite} />
      {!hideLocales && (
        <div className={styles.locale}>
          <Locales locale={locale} linkRenderer={linkRenderer} />
        </div>
      )}
    </div>
  </Hidden>);
}

GlobalNav.displayName = 'GlobalNav';

GlobalNav.propTypes = {
  locale: PropTypes.oneOf(['AU', 'NZ']).isRequired,
  linkRenderer: PropTypes.func.isRequired,
  hideLocales: PropTypes.bool,
  activePartnerSite: PropTypes.oneOf(['jobs', 'courses', 'businesses', 'volunteering'])
};

GlobalNav.defaultProps = {
  hideLocales: false,
  activePartnerSite: null
};
