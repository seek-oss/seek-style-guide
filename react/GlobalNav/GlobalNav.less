@import (reference) "~seek-style-guide/theme";

.root {
  order: 0;
  width: 100%;
  background-color: @sk-background;
  padding: 0 (@gutter-width / 2);
}

.content {
  position: relative;
  display: flex;
  flex-grow: 1;
  justify-content: center;
  max-width: @container-width;
  margin: 0 auto;
}

.locale {
  position: absolute;
  right: 0;
}
