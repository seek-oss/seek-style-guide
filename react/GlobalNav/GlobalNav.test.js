import React from 'react';
import { render } from 'enzyme';

import GlobalNav from './GlobalNav';

const renderGlobalNav = props => render(<GlobalNav {...props} />);
const linkRenderer = props => (<a {...props} />);

describe('GlobalNav:', () => {
  describe('locale states:', () => {
    it('should render with locale of AU', () => {
      expect(renderGlobalNav({ locale: 'AU', linkRenderer, activePartnerSite: 'jobs' })).toMatchSnapshot();
    });

    it('should render with locale of NZ', () => {
      expect(renderGlobalNav({ locale: 'NZ', linkRenderer, activePartnerSite: 'jobs' })).toMatchSnapshot();
    });

    it('should render without locales if specified', () => {
      expect(renderGlobalNav({ locale: 'NZ', linkRenderer, activePartnerSite: 'jobs', hideLocales: true })).toMatchSnapshot();
    });
  });
  describe('active partner site:', () => {
    it('should render with jobs link as active', () => {
      expect(renderGlobalNav({ locale: 'AU', linkRenderer, activePartnerSite: 'jobs' })).toMatchSnapshot();
    });
    it('should render with courses link as active', () => {
      expect(renderGlobalNav({ locale: 'AU', linkRenderer, activePartnerSite: 'courses' })).toMatchSnapshot();
    });
    it('should render with businesses link as active', () => {
      expect(renderGlobalNav({ locale: 'AU', linkRenderer, activePartnerSite: 'businesses' })).toMatchSnapshot();
    });
    it('should render with volunteering link as active', () => {
      expect(renderGlobalNav({ locale: 'AU', linkRenderer, activePartnerSite: 'volunteering' })).toMatchSnapshot();
    });
  });
});
