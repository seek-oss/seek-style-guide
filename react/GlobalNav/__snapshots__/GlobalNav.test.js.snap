// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalNav: active partner states: should render with businesses link as active 1`] = `
<span
  class="root mobile print"
>
  <div
    class="content"
  >
    <nav
      aria-labelledby="PartnerSites"
      role="navigation"
    >
      <span
        class="root"
      >
        <h1
          id="PartnerSites"
        >
          Partner Sites
        </h1>
      </span>
      <ul
        class="list"
      >
        <li>
          <a
            class="link link_isJobs"
            data-analytics="header:jobs"
            href="https://www.seek.com.au"
            title="SEEK Jobs"
          >
            Jobs
          </a>
        </li>
        <li>
          <a
            class="link link_isLearning"
            data-analytics="header:courses"
            href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
            title="SEEK Learning"
          >
            Courses
          </a>
        </li>
        <li>
          <span
            class="link link_isActive"
            title="SEEK Business"
          >
            Businesses for sale
          </span>
        </li>
        <li>
          <a
            class="link link_isVolunteering"
            data-analytics="header:volunteering"
            href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
            title="SEEK Volunteer"
          >
            Volunteering
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="locale"
    >
      <nav
        aria-labelledby="Locales"
        class="root"
        role="navigation"
      >
        <span
          class="root"
        >
          <h1
            id="Locales"
          >
            Select your country
          </h1>
        </span>
        <ul
          class="list"
        >
          <li
            class="listItem"
          >
            <span
              class="locale_isActive"
            >
              AU
            </span>
          </li>
          <li
            class="listItem"
          >
            <a
              class="localeLink"
              data-analytics="header:nz+homepage"
              href="https://www.seek.co.nz"
              title="SEEK New Zealand"
            >
              NZ
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</span>
`;

exports[`GlobalNav: active partner states: should render with courses link as active 1`] = `
<span
  class="root mobile print"
>
  <div
    class="content"
  >
    <nav
      aria-labelledby="PartnerSites"
      role="navigation"
    >
      <span
        class="root"
      >
        <h1
          id="PartnerSites"
        >
          Partner Sites
        </h1>
      </span>
      <ul
        class="list"
      >
        <li>
          <a
            class="link link_isJobs"
            data-analytics="header:jobs"
            href="https://www.seek.com.au"
            title="SEEK Jobs"
          >
            Jobs
          </a>
        </li>
        <li>
          <span
            class="link link_isActive"
            title="SEEK Learning"
          >
            Courses
          </span>
        </li>
        <li>
          <a
            class="link link_isBusiness"
            data-analytics="header:business+for+sale"
            href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
            title="SEEK Business"
          >
            Businesses for sale
          </a>
        </li>
        <li>
          <a
            class="link link_isVolunteering"
            data-analytics="header:volunteering"
            href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
            title="SEEK Volunteer"
          >
            Volunteering
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="locale"
    >
      <nav
        aria-labelledby="Locales"
        class="root"
        role="navigation"
      >
        <span
          class="root"
        >
          <h1
            id="Locales"
          >
            Select your country
          </h1>
        </span>
        <ul
          class="list"
        >
          <li
            class="listItem"
          >
            <span
              class="locale_isActive"
            >
              AU
            </span>
          </li>
          <li
            class="listItem"
          >
            <a
              class="localeLink"
              data-analytics="header:nz+homepage"
              href="https://www.seek.co.nz"
              title="SEEK New Zealand"
            >
              NZ
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</span>
`;

exports[`GlobalNav: active partner states: should render with jobs link as active 1`] = `
<span
  class="root mobile print"
>
  <div
    class="content"
  >
    <nav
      aria-labelledby="PartnerSites"
      role="navigation"
    >
      <span
        class="root"
      >
        <h1
          id="PartnerSites"
        >
          Partner Sites
        </h1>
      </span>
      <ul
        class="list"
      >
        <li>
          <span
            class="link link_isActive"
            title="SEEK Jobs"
          >
            Jobs
          </span>
        </li>
        <li>
          <a
            class="link link_isLearning"
            data-analytics="header:courses"
            href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
            title="SEEK Learning"
          >
            Courses
          </a>
        </li>
        <li>
          <a
            class="link link_isBusiness"
            data-analytics="header:business+for+sale"
            href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
            title="SEEK Business"
          >
            Businesses for sale
          </a>
        </li>
        <li>
          <a
            class="link link_isVolunteering"
            data-analytics="header:volunteering"
            href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
            title="SEEK Volunteer"
          >
            Volunteering
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="locale"
    >
      <nav
        aria-labelledby="Locales"
        class="root"
        role="navigation"
      >
        <span
          class="root"
        >
          <h1
            id="Locales"
          >
            Select your country
          </h1>
        </span>
        <ul
          class="list"
        >
          <li
            class="listItem"
          >
            <span
              class="locale_isActive"
            >
              AU
            </span>
          </li>
          <li
            class="listItem"
          >
            <a
              class="localeLink"
              data-analytics="header:nz+homepage"
              href="https://www.seek.co.nz"
              title="SEEK New Zealand"
            >
              NZ
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</span>
`;

exports[`GlobalNav: active partner states: should render with volunteering link as active 1`] = `
<span
  class="root mobile print"
>
  <div
    class="content"
  >
    <nav
      aria-labelledby="PartnerSites"
      role="navigation"
    >
      <span
        class="root"
      >
        <h1
          id="PartnerSites"
        >
          Partner Sites
        </h1>
      </span>
      <ul
        class="list"
      >
        <li>
          <a
            class="link link_isJobs"
            data-analytics="header:jobs"
            href="https://www.seek.com.au"
            title="SEEK Jobs"
          >
            Jobs
          </a>
        </li>
        <li>
          <a
            class="link link_isLearning"
            data-analytics="header:courses"
            href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
            title="SEEK Learning"
          >
            Courses
          </a>
        </li>
        <li>
          <a
            class="link link_isBusiness"
            data-analytics="header:business+for+sale"
            href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
            title="SEEK Business"
          >
            Businesses for sale
          </a>
        </li>
        <li>
          <span
            class="link link_isActive"
            title="SEEK Volunteer"
          >
            Volunteering
          </span>
        </li>
      </ul>
    </nav>
    <div
      class="locale"
    >
      <nav
        aria-labelledby="Locales"
        class="root"
        role="navigation"
      >
        <span
          class="root"
        >
          <h1
            id="Locales"
          >
            Select your country
          </h1>
        </span>
        <ul
          class="list"
        >
          <li
            class="listItem"
          >
            <span
              class="locale_isActive"
            >
              AU
            </span>
          </li>
          <li
            class="listItem"
          >
            <a
              class="localeLink"
              data-analytics="header:nz+homepage"
              href="https://www.seek.co.nz"
              title="SEEK New Zealand"
            >
              NZ
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</span>
`;

exports[`GlobalNav: locale states: should render with locale of AU 1`] = `
<span
  class="root mobile print"
>
  <div
    class="content"
  >
    <nav
      aria-labelledby="PartnerSites"
      role="navigation"
    >
      <span
        class="root"
      >
        <h1
          id="PartnerSites"
        >
          Partner Sites
        </h1>
      </span>
      <ul
        class="list"
      >
        <li>
          <span
            class="link link_isActive"
            title="SEEK Jobs"
          >
            Jobs
          </span>
        </li>
        <li>
          <a
            class="link link_isLearning"
            data-analytics="header:courses"
            href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
            title="SEEK Learning"
          >
            Courses
          </a>
        </li>
        <li>
          <a
            class="link link_isBusiness"
            data-analytics="header:business+for+sale"
            href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
            title="SEEK Business"
          >
            Businesses for sale
          </a>
        </li>
        <li>
          <a
            class="link link_isVolunteering"
            data-analytics="header:volunteering"
            href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
            title="SEEK Volunteer"
          >
            Volunteering
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="locale"
    >
      <nav
        aria-labelledby="Locales"
        class="root"
        role="navigation"
      >
        <span
          class="root"
        >
          <h1
            id="Locales"
          >
            Select your country
          </h1>
        </span>
        <ul
          class="list"
        >
          <li
            class="listItem"
          >
            <span
              class="locale_isActive"
            >
              AU
            </span>
          </li>
          <li
            class="listItem"
          >
            <a
              class="localeLink"
              data-analytics="header:nz+homepage"
              href="https://www.seek.co.nz"
              title="SEEK New Zealand"
            >
              NZ
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</span>
`;

exports[`GlobalNav: locale states: should render with locale of NZ 1`] = `
<span
  class="root mobile print"
>
  <div
    class="content"
  >
    <nav
      aria-labelledby="PartnerSites"
      role="navigation"
    >
      <span
        class="root"
      >
        <h1
          id="PartnerSites"
        >
          Partner Sites
        </h1>
      </span>
      <ul
        class="list"
      >
        <li>
          <span
            class="link link_isActive"
            title="SEEK Jobs"
          >
            Jobs
          </span>
        </li>
        <li>
          <a
            class="link link_isLearning"
            data-analytics="header:courses"
            href="https://www.seeklearning.co.nz/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
            title="SEEK Learning"
          >
            Courses
          </a>
        </li>
        <li>
          <a
            class="link link_isBusiness"
            data-analytics="header:business+for+sale"
            href="https://www.seekbusiness.com.au/?site=nz&tracking=sk:main:nz:nav:bus"
            title="SEEK Business"
          >
            Businesses for sale
          </a>
        </li>
        <li>
          <a
            class="link link_isVolunteering"
            data-analytics="header:volunteering"
            href="https://seekvolunteer.co.nz/?tracking=SKMNZ:main+header"
            title="SEEK Volunteer"
          >
            Volunteering
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="locale"
    >
      <nav
        aria-labelledby="Locales"
        class="root"
        role="navigation"
      >
        <span
          class="root"
        >
          <h1
            id="Locales"
          >
            Select your country
          </h1>
        </span>
        <ul
          class="list list_isNZ"
        >
          <li
            class="listItem"
          >
            <a
              class="localeLink"
              data-analytics="header:au+homepage"
              href="https://www.seek.com.au"
              title="SEEK Australia"
            >
              AU
            </a>
          </li>
          <li
            class="listItem"
          >
            <span
              class="locale_isActive"
            >
              NZ
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</span>
`;

exports[`GlobalNav: locale states: should render without locales if specified 1`] = `
<span
  class="root mobile print"
>
  <div
    class="content"
  >
    <nav
      aria-labelledby="PartnerSites"
      role="navigation"
    >
      <span
        class="root"
      >
        <h1
          id="PartnerSites"
        >
          Partner Sites
        </h1>
      </span>
      <ul
        class="list"
      >
        <li>
          <span
            class="link link_isActive"
            title="SEEK Jobs"
          >
            Jobs
          </span>
        </li>
        <li>
          <a
            class="link link_isLearning"
            data-analytics="header:courses"
            href="https://www.seeklearning.co.nz/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
            title="SEEK Learning"
          >
            Courses
          </a>
        </li>
        <li>
          <a
            class="link link_isBusiness"
            data-analytics="header:business+for+sale"
            href="https://www.seekbusiness.com.au/?site=nz&tracking=sk:main:nz:nav:bus"
            title="SEEK Business"
          >
            Businesses for sale
          </a>
        </li>
        <li>
          <a
            class="link link_isVolunteering"
            data-analytics="header:volunteering"
            href="https://seekvolunteer.co.nz/?tracking=SKMNZ:main+header"
            title="SEEK Volunteer"
          >
            Volunteering
          </a>
        </li>
      </ul>
    </nav>
  </div>
</span>
`;
